cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -596,25 +600,35 @@ public void markApplicationsLatest(Collection<ApplicationId> appIds)
   @Override
   public int addLatestApplication(ApplicationId id, ApplicationMeta meta) throws ConflictException {
     return TransactionRunners.run(transactionRunner, context -> {
-      return getAppMetadataStore(context).createLatestApplicationVersion(id, meta);
+      WriteApplicationResponse response = getAppMetadataStore(
+          context).createLatestApplicationVersion(id, meta);
+      getSourceControlMetadataStore(context).setNamespaceSourceControlMeta(id,
+          response.getSourceControlMeta());
+      return response.getApplicationEditNumber();
     }, ConflictException.class);
   }
 
   @Override
-  public int addApplication(ApplicationId id, ApplicationMeta meta, boolean isLatest) throws ConflictException {
+  public int addApplication(ApplicationId id, ApplicationMeta meta, boolean isLatest)
+      throws ConflictException {
     return TransactionRunners.run(transactionRunner, context -> {
-      return getAppMetadataStore(context).createApplicationVersion(id, meta, isLatest);
+      WriteApplicationResponse response = getAppMetadataStore(context).createApplicationVersion(id,
```

Comment:
I think ApplicationMeta already has source control meta. So you do not need to use a intermediary response object `WriteApplicationResponse`  here

Reply:
Right. I have made the change. Thank you